packages/compass-web/src/connection-storage.spec.ts

```diff
@@ -172,4 +176,133 @@ describe('buildConnectionInfoFromClusterDescription', function () {
         });
     });
   }
+
+  it('should throw if deployment item is missing', function () {
+    try {
+      buildConnectionInfoFromClusterDescription(
+        'ws://test',
+        '123',
+        'abc',
+        {
+          '@provider': 'mock',
+          uniqueId: 'abc',
+          groupId: 'abc',
+          name: 'Cluster0',
+          clusterType: 'REPLICASET',
+          srvAddress: 'test',
+          state: 'test',
+          deploymentItemName: 'test',
+          dataProcessingRegion: { regionalUrl: 'test' },
+        },
+        deployment
+      );
+      expect.fail('Expected method to throw');
+    } catch (err) {
+      expect(err).to.have.property(
+        'message',
+        "Can't build metrics info when deployment item is not found"
+      );
+    }
+  });
+});
+
+describe('AtlasCloudConnectionStorage', function () {
+  const testClusters: Record<
+    string,
+    Partial<ClusterDescriptionWithDataProcessingRegion>
+  > = {
+    Cluster0: {
+      '@provider': 'AWS',
+      groupId: 'abc',
+      name: 'Cluster0',
+      clusterType: 'REPLICASET',
+      srvAddress: 'test',
+      state: 'test',
+      deploymentItemName: 'replicaSet-xxx',
+      dataProcessingRegion: { regionalUrl: 'test' },
+    },
+    NoDeploymentItem: {
+      '@provider': 'AWS',
+      groupId: 'abc',
+      name: 'NoDeploymentItem',
+      clusterType: 'REPLICASET',
+      srvAddress: 'test',
+      state: 'test',
+      deploymentItemName: 'not-found',
+      dataProcessingRegion: { regionalUrl: 'test' },
+    },
+    NoSrvAddress: {
+      '@provider': 'AWS',
+      name: 'NoSrvAddress',
+    },
+    Paused: {
+      '@provider': 'AWS',
+      name: 'Paused',
+      isPaused: true,
+    },
+    WillThrowOnFetch: {
+      '@provider': 'AWS',
+      name: 'WillThrowOnFetch',
+    },
+  };
+
+  describe('#loadAll', function () {
+    it('should load connection descriptions filtering out the ones that failed to fetch', async function () {
+      const atlasService = {
+        cloudEndpoint(path: string) {
+          return path;
+        },
+        driverProxyEndpoint(path: string) {
+          return path;
+        },
+        authenticatedFetch(path: string) {
+          let payload: any;
+          if (path === '/deployment/abc') {
+            payload = deployment;
+          }
+          if (path === '/nds/clusters/abc') {
+            payload = Array.from(Object.values(testClusters));
+          }
+          const { groups } =
+            /\/nds\/clusters\/abc\/(?<clusterName>.+?)\/+?/.exec(path) ?? {
```

Comment:
```suggestion
            /^\/nds\/clusters\/abc\/(?<clusterName>.+?)\/+?$/.exec(path) ?? {
```

(I think this would be correct here? Not blocking for sure, the current version works, it just feels safer to use an anchored RE)

Reply:
(and I also forgot the "any char" before the end of string)

Reply:
Ah! Yeah, that also helps, because I had figured the previous RE was intended to check that the string ends with a slash :)